myia/prim/py_implementations.py

```diff
@@ -146,6 +146,22 @@ def bool_not(x):
     return not x
 
 
+@register(primops.bool_and)
+def bool_and(x, y):
+    """Implement `bool_and`."""
+    assert x is True or x is False
+    assert y is True or y is False
```

Comment:
Why not? `bool_and` is only meant to work on boolean values.

Reply:
Yes, but we don't do checks like that for any other primitive.

Reply:
We do now. The previous PR added a call to `_assert_scalar(x, y)` to every scalar primitive (which didn't break anything, because they were never called on non-scalars to begin with), and `assert x is True or x is False` in also present in `bool_not`.

Reply:
I missed that.  So in that case, this is good.